prototypes/pipe-clamps.lua

```diff
@@ -161,6 +161,42 @@ local function build_picture_table(current_picture_table, current_name)
     }
 end
 
+local function build_composite_icon(current_entity)
+    local new_icons = {}
+
+    if current_entity.icons then
+        -- Icons table, transcribe layers into new_icons, fully define the IconData
+        for _, icon_data in pairs(current_entity.icons) do
+            table.insert(new_icons, {
+                icon = icon_data.icon,
+                icon_size = icon_data.icon_size or current_entity.icon_size,
+                icon_mipmaps = icon_data.icon_mipmaps or current_entity.icon_mipmaps,
+                tint = icon_data.tint,
+                shift = icon_data.shift,
+                scale = icon_data.scale or 1,
+            })
+        end
+    else
+        -- Standard icon, convert to an icons definition
+        table.insert(new_icons, {
```

Comment:
Same

Reply:
Additionally there should be a check to make sure there is a .icon defined here. Not that it matters for this instance. But eventually this useful function will be moved to Stdlib Data library :)

Reply:
I don't think a check to ensure a .icon is defined is necessary. If icons is not defined, **and icon** is not defined, the game itself will throw an error, for example:

![image](https://user-images.githubusercontent.com/35714562/93948110-641b0900-fcf2-11ea-92d0-f9860bf2719b.png)

Ergo, if there isn't an icons definition, there should be an icon definition, and though we could protect against their not being an icon definition _as well_, the game won't load anyways in that case.

Reply:
hmmmm yes I suppose you are right.